gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/HiveDatasetDescriptor.java

```diff
@@ -102,25 +112,33 @@ protected boolean isPathContaining(DatasetDescriptor other) {
     String otherDbName = parts.get(0);
     String otherTableNames = parts.get(1);
 
-    if(!Pattern.compile(this.databaseName).matcher(otherDbName).matches()) {
+    if (!isDbAccepted(otherDbName)) {
       return false;
     }
 
     List<String> tables = Splitter.on(",").splitToList(otherTableNames);
     for (String table : tables) {
-      if (!isPathContaining(table)) {
+      if (!isTableAccepted(table)) {
         return false;
       }
     }
     return true;
   }
 
-  private boolean isPathContaining(String tableName) {
-    if (tableName == null) {
-      return false;
+  private boolean isDbAccepted(String otherDbName) {
+    List<String> dbs = Splitter.on(",").splitToList(this.databaseName);
+
+    for (String db : dbs) {
+      if (new GlobPattern(db).matches(otherDbName)) {
```

Comment:
Should we use java.util.regex patterns for contains() checks, instead of Hadoop GlobPattern which is more restrictive.

Reply:
replaced it with with WhitelistBlacklist, which internally uses java.util.regex patterns